gobblin-runtime/src/main/java/org/apache/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -771,6 +780,20 @@ protected void setup(Context context) {
         gobblinJobState.setProp(entry.getKey(), entry.getValue().unwrapped().toString());
       }
 
+      // add some more MR task related configs
+
+      String[] tokens = taskAttemptId.toString().split("_");
+      TaskType taskType = taskAttemptId.getTaskType();
+      gobblinJobState.setProp(MR_TYPE_KEY, taskType.name());
+
+      if (taskType.equals(TaskType.MAP)) {
+        gobblinJobState.setProp(MAPPER_TASK_NUM_KEY, tokens[4]);
```

Comment:
Add a check for the length of the "tokens" array or an assertion that tokens should be of length 6. Further, you may want to access the map/reduce task num as tokens[tokens.length - 1].